main/inc/lib/attendance.lib.php

```diff
@@ -2256,4 +2293,123 @@ public function exportAttendanceLogin($startDate, $endDate)
         $pdf = new PDF('A4', null, $params);
         $pdf->html_to_pdf_with_template($tableToString);
     }
+
+    /**
+     * Return all course for a student between dates and order by key date (Y-m-d).
+     *
+     * @param int  $student_id
+     * @param bool $orderDesc
+     *
+     * @return array
+     */
+    public function getCoursesWithAttendance(
+        $student_id = 0,
+        DateTime $startDate = null,
+        DateTime $endDate = null,
+        $orderDesc = true
+    ) {
+        $attendanceLib = new Attendance();
+        $data = [];
+        // unused by now
+        /*
+         $specialCourses = CourseManager::returnSpecialCourses(
+            $student_id, false, false
+        );
+         */
+        $courses = CourseManager::returnCourses(
+            $student_id,
+            false,
+            false
+        );
+
+        /* Get course with (in_category) and without (not_category) category */
+        foreach ($courses as $coursesKey => $courseData) {
+            /*
+             * $coursesKey can be in_category or not_category for courses
+             * */
+            $totalCoursesNoCategory = count($courseData);
+            for ($i = 0; $i < $totalCoursesNoCategory; $i++) {
+                $courseItem = $courseData[$i];
+                /* Get all attendance by courses*/
+                $attenances = $attendanceLib->get_attendances_list($courseItem['course_id']);
+                $temp = [];
+                $sheetsProccessed = [];
+                $tempDate = [];
+                foreach ($attenances as $attendanceData) {
+                    $attendance_id = $attendanceData['id'];
+                    $course_id = $courseItem['course_id'];
```

Comment:
This is new code so $attendanceId and $courseId (camel case). Non camel-case elements are reserved for:
- array indexes
- language variables (starting with an uppercase letter)
- old variables not yes converted